neuraxle/metaopt/random.py

```diff
@@ -114,6 +117,278 @@ def _split(self, data_inputs):
         return splitted_data_inputs
 
 
+class WalkForwardTimeSeriesCrossValidation(BaseCrossValidation):
+    """
+    This will change perform a classic walk forward cross validation by performing a forward rolling split.
+    All the training split will have the same `validation_window_size` size. The start and end of each training split
+    will increase at the same time toward the end at each forward split. Same principle apply with the validation split,
+    where the start and end will increase in the same manner toward the end. Each validation split start after a certain
+    time delay (if padding is set) after their corresponding training split.
+
+    The difference in start position between two consecutive validation split are equal to one `validation_window_size`.
+
+    Notes: The data supported by this cross validation is nd.array of shape = (batch, timeSeries, ...) .
+    The array can have an arbitrary number of dimension, but the time series axis is currently limited to `axis=1`.
+    Recommended format for the data is an numpy array of shape [batch_size, total_time_steps, n_features].
+
+    :param validation_window_size: the window size of each validation split and
+        also the time step taken between each forward roll, by default 3.
+    :param training_window_size: the window size of training split by default None.
+        If None : It takes the value `validation_window_size`.
+    :param padding_between_training_and_validation: the size of the padding between the end of the training split
+        and the start of the validation split, by default 0.
+    :param drop_remainder: drop the last split if the last validation split does not coincide
+        with a full validation_window_size, by default False.
+    :param scoring_function: scoring function use to validate performance if it is not None, by default r2_score,
+    :param joiner the joiner callable that can join the different result together.
+    :return: WalkForwardTimeSeriesCrossValidation instance.
+    """
+
+    def __init__(self, validation_window_size, training_window_size=None, padding_between_training_and_validation=0,
```

Comment:
I would suggest inverting the order of `validation_window_size, training_window_size=None` to `training_window_size, validation_window_size=None` simply because train comes before validation in natural thinking order.

Reply:
I have made the change.